source/dhtslib/sam.d

```diff
@@ -363,7 +364,7 @@ struct SAMFile
             //auto close=hclose(this.f);
             //if (ret != 0) writeln("There was an error hfile");
         }
-        else
+        else if(this.fp !is null)
```

Comment:
does this fix all of the concerns in the TODO:hts_close segfaults and "Causes double free" comments?
If so, remove comments. Also, if nothing in if {} body, remove and make only if(this.fp !is null) block.

Reply:
also i note the logic was reversed, so it should have always (?) segfaulted.

are you testing this block, including passing existing FILE and closing it from here?